docs/source/06_nodes_and_pipelines/03_modular_pipelines.md

```diff
@@ -4,406 +4,267 @@
 
-```bash
-kedro pipeline pull <path-to-your-project-root>/src/dist/<pipeline_name>-0.1-py3-none-any.whl
-```
+### Ensuring portability
 
-- Pulling a modular pipeline from S3:
+Kedro modular pipelines are shareable between Kedro codebases via [micro-packaging](04_micro_packaging.md), but you need to follow a couple of rules to ensure portability:
 
-```bash
-kedro pipeline pull https://<bucket_name>.s3.<aws-region>.amazonaws.com/<pipeline_name>-0.1-py3-none-any.whl
-```
-
-- Pulling a modular pipeline from PyPI:
+* Modular pipelines should use [relative imports](https://docs.python.org/3/reference/import.html#package-relative-imports) (as opposed to 'absolute' imports) when accessing their own objects, but are free to use absolute imports otherwise.
+* Modular pipelines should **not** depend on the main Python package as this would break portability to another project.
+* Catalog references are not packaged when sharing/consuming modular pipelines.
+* Kedro will only look for top level configuration `/conf/` providing a configuration folder within the pipeline folder will have no effect.
+* It is recommended that you document the configuration required (parameters and catalog) in the local `README.md` file for any downstream consumers.
 
-```bash
-kedro pipeline pull <pypi-package-name>
-```
+### Providing modular pipeline specific dependencies
 
-If you are pulling the pipeline from a location that isn't PyPI, Kedro uses [`fsspec`](https://filesystem-spec.readthedocs.io/en/latest/) to locate and pull down your pipeline. If you need to provide any `fsspec`-specific arguments (say, if you're pulling your pipeline down from an S3 bucket and want to provide the S3 credentials inline or from a local server that requires tokens in the header) then you can use the `--fs-args` option to point to a YAML (or any `anyconfig`-supported configuration) file that contains the required configuration.
+* A modular pipeline **may** have external dependencies specified in a local `requirements.txt` file.
+* Pipeline specific dependencies are scooped up during [micro-packaging](04_micro_packaging.md) process.
+* These dependencies are _not_ currently installed by the [`kedro install`](../09_development/03_commands_reference.md#install-all-package-dependencies) command and will have to be manually installed.
 
-```bash
-kedro pipeline pull https://<url-to-pipeline.whl> --fs-args pipeline_pull_args.yml
-```
+## Using the modular `pipeline()` wrapper to provide overrides
 
-where
+This wrapper really unlocks the power of modular pipelines.
 
-```
-# pipeline_pull_args.yml
-client_kwargs:
-  headers:
-    Authorization: token <token>
-```
+* It allows you to start instantiating the same pipeline multiple times.
+* These will be static in terms of structure, but dynamic in terms of `inputs`/`outputs`/`parameters`.
+* It also allows you to simplify both your mental models and Kedro-Viz visualisations via `namespace`s.
```

Comment:
```suggestion
* It also allows you to simplify both your mental models and Kedro-Viz visualisations via `namespaces`.
```

Reply:
@idanov I think this is covered later in more detail